src/zenml/integrations/smtp_email/steps/smtp_email_alerter_post_step.py

```diff
@@ -0,0 +1,70 @@
+#  Copyright (c) ZenML GmbH 2022. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+"""Implementation of a step that sends an email alert."""
+
+from typing import Optional
+
+from zenml import get_step_context, step
+from zenml.client import Client
+from zenml.integrations.smtp_email.alerters.smtp_email_alerter import (
+    SMTPEmailAlerterParameters,
+    SMTPEmailAlerterPayload,
+)
+
+
+@step
+def smtp_email_alerter_post_step(
+    message: str,
+    recipient_email: Optional[str] = None,
+    subject: Optional[str] = None,
+    include_pipeline_info: bool = True,
+) -> None:
+    """Step that sends an email alert using the active alerter.
+
+    Args:
+        message: Message to be included in the email.
+        recipient_email: Optional recipient email address. If not provided,
+            the email will be sent to the default recipient configured
+            in the alerter.
+        subject: Optional custom subject for the email.
+        include_pipeline_info: If True, includes pipeline and step information
+            in the email.
+
+    Raises:
+        RuntimeError: If no alerter is configured in the active stack.
+    """
+    alerter = Client().active_stack.alerter
+    if not alerter:
```

Comment:
Shouldn't we also check the alerter type here?

In general it feels kinda weird to me what is happening with alerters. Shouldn't there be **one** alerter post step, which works independently of the alerter type? This is the whole reason for stack switching.

I would suggest that we think of a generic format that get's passed to `alerter.post(...)`, and we can then get rid of all these specific steps. We can also avoid checks in the open AI hooks for example that handle the email alerter separately. The alerter itself should be responsible for the formatting, not some outside logic IMO.

Reply:
I've heavily updated it based on this comment. Hope I haven't gone too far. There's no breaking changes, but it's def a fairly large change.